src/lti/LTI_Message_Launch.php

```diff
@@ -205,7 +211,7 @@ public function get_launch_id() {
         return $this->launch_id;
     }
 
-    private function get_public_key() {
+    protected function get_public_key() {

```

Comment:
why?

Reply:
Because otherwise you have to get the keyset from the key_set_url. Making this protected lets us stub that out.